__tests__/schema.test.ts

```diff
@@ -0,0 +1,187 @@
+		const email = "user@email.com";
+		await deleteSupabaseUser(email); // clean up before running
+		const { data, error } = await supabaseAnonClient.auth.signUp({
+			email: email,
+			password: "12345678",
+		});
+		expect(error).toBeNull();
+		expect(data).toBeDefined();
+		const { data: trees, error: treesError } = await supabaseAnonClient
+			.from("trees")
+			.select("*")
+			.limit(10);
+		expect(treesError).toBeNull();
+		expect(trees).toHaveLength(10);
+
+		const { data: adoptedTrees, error: adoptedTreesError } =
+			await supabaseServiceRoleClient
+				.from("trees_adopted")
+				.insert(
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					trees!.map((tree) => ({
+						uuid: data.user?.id,
+						tree_id: tree.id,
+					}))
+				)
+				.select("*");
+		expect(adoptedTreesError).toBeNull();
+		expect(adoptedTrees).toHaveLength(10);
+		const { data: userTrees, error: userTreesError } =
+			await supabaseServiceRoleClient
+				.from("trees_watered")
+				.insert(
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					trees!.map((tree) => ({
+						uuid: data.user?.id,
+						amount: 1,
+						timestamp: new Date().toISOString(),
+						username: "user",
+						tree_id: tree.id,
+					}))
+				)
+				.select("*");
+		expect(userTreesError).toBeNull();
+		expect(userTrees).toHaveLength(10);
+
+		// since wa can not pass the token to our supabase client, we need to use fetch directly
+		const response = await fetch(`${SUPABASE_URL}/rest/v1/rpc/remove_account`, {
+			method: "POST",
+			headers: {
+				apikey: SUPABASE_ANON_KEY,
+				"Content-Type": "application/json",
+				Authorization: `Bearer ${data.session?.access_token}`,
+			},
+		});
+		expect(response.ok).toBeTruthy();
+		expect(response.status).toBe(204);
+		const { data: treesAfter, error: treesAfterError } =
+			await supabaseAnonClient
+				.from("trees_watered")
+				.select("*")
+				.eq("uuid", data.user?.id);
+		expect(treesAfterError).toBeNull();
+		expect(treesAfter).toHaveLength(0);
+
+		const { data: adoptedTreesAfter, error: adoptedTreesAfterError } =
+			await supabaseAnonClient
+				.from("trees_adopted")
+				.select("*")
+				.eq("uuid", data.user?.id);
+		expect(adoptedTreesAfterError).toBeNull();
+		expect(adoptedTreesAfter).toHaveLength(0);
+		await truncateTreesWaterd();
+	});
+
+	test("if a user changes his username all the usernames on the trees_watered table should change too", async () => {
+		const email = "foo@bar.com";
+		await deleteSupabaseUser(email);
+		await truncateTreesWaterd();
+		const { data, error } = await supabaseAnonClient.auth.signUp({
+			email: email,
+			password: "12345678",
+		});
+		expect(error).toBeNull();
+		expect(data).toBeDefined();
+		const { data: trees, error: treesError } = await supabaseAnonClient
+			.from("trees")
+			.select("*")
+			.limit(10);
+		expect(treesError).toBeNull();
+		expect(trees).toHaveLength(10);
```

Comment:
I've seen the 10 show up multiple times. Where does this number come from? Is there a way to make this dynamic? So that we set what I suspect are mock trees and fetch their length somehow?

Reply:
Yeah. Its a magic number. I have 14 trees in the seed. 10 random ones and 4 in front of the Citylab. I could of course make this dynamic, but didn't find it that important.

Reply:
Made it dynamic.